src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWalletActor.scala

```diff
@@ -420,63 +427,68 @@ class ErgoWalletActor(settings: ErgoSettings, boxSelector: BoxSelector)
   }
 
   /**
-    * Generates new transaction according to a given requests using available boxes.
+    * Generates new transaction according to given requests using available or provided boxes.
+    *
+    * @param requests - requests to transfer funds or to issue an asset
+    * @param inputsRaw - user-provided inputs. If empty then wallet is looking for inputs itself. If non-empty, then
+    *                  the wallet is not adding anything, thus the user in this case should take care about satisfying
+    *                  the (sum(inputs) == sum(outputs)) preservation rule for ergs.
+    * @param dataInputsRaw - user-provided data (read-only) inputs. Wallet is not able to provide data inputs
+    *                      (if they are needed in order to spend the spendable inputs).
+    *
+    * @return generated transaction or an error
     */
   private def generateTransactionWithOutputs(requests: Seq[TransactionGenerationRequest],
                                              inputsRaw: Seq[String],
                                              dataInputsRaw: Seq[String]): Try[ErgoTransaction] = Try {
+
+    // A helper which converts Base16-encoded boxes to ErgoBox instances
+    def stringsToBoxes(strings: Seq[String]): Seq[ErgoBox] =
+      strings.map(in => Base16.decode(in).flatMap(ErgoBoxSerializer.parseBytesTry)).map(_.get)
```

Comment:
Use `Seq(Try()).collect { case Success(x) => x }` at least.

Reply:
@oskin1 generateTransactionWithOutputs body is wrapped in Try, so .get isn't a problem